### ISSUE TITLE
Merge conditional branches on ObjectLike arrays similarly to variables and properties?

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#456]
Low priority, this is rarely an issue, and I can work around it by `$data['field'] = $data['field'] ?? [];`

Observed: PossiblyNullIterator
Expected: No warning, the field was checked for null.

I assume that this happens because properties are treated the same way as variables in the local scope (E.g. entries exist for both $obj and '$obj->prop, but ObjectLike can't be narrowed.

```php
<?php

function test() : void {
    $data = [
        'field' => (rand() % 2 > 0 ? ['value'] : null)
    ];
    if (is_null($data['field'])) {
        $data['field'] = [];
    }  // Psalm doesn't currently infer $data['field'] is non-null.
    foreach ($data['field'] as $value) {  // PossiblyNullIterator - src/iterator.php:10:14 - Cannot iterate over nullable var array{0:string}|null
        echo $value;
    }
}
```

**Comment by muglug** [comment]
Sure, low priority, but this is most definitely a bug

**Comment by muglug** [comment]
This has been fixed at some point...
